extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxRequestHandler.java

```diff
@@ -96,12 +97,16 @@ private void dispatch(RoutingContext routingContext, InputStream is, VertxOutput
             HttpServerResponse response = request.response();
             VertxHttpResponse vertxResponse = new VertxHttpResponse(request, dispatcher.getProviderFactory(),
                     request.method(), allocator, output);
-            // client address may not be available with VirtualHttp
-            SocketAddress socketAddress = request.remoteAddress();
-            String host = socketAddress != null ? socketAddress.host() : null;
+            // client address may not be available with VirtualHttp;
+            // using a supplier to make the remote Address resolution lazy: often it's not needed.
+            Supplier<String> hostNameProvider = () -> {
```

Comment:
We try and avoid lambdas in the runtime part of the code as they are much more costly in terms of metaspsace than an anonymous class. 

If this is expensive shouldn't the result of this be cached either in the Supplier or in VertxHttpRequest in case getRemoteAddress is called multiple times?

Reply:
ok I can change the lambda to an anonymous class. I know the metaspace cost, TBH I gave up on avoiding them in runtime code as we have many of them anyway, I had the impression nobody except me was still following that recommendation :)

It's not clear to me if it's worth caching the result. I can do that "just in case", but FYI the getter is not invoked during hot-path code in the benchmarks I've run: I verified this by initially running a modified version which was throing an exception.
I had to remove the exception-throwing code as some tests would fail, but it seems seldomly used - I don't know for sure though I guess I could have seen an happy path in my runs.

Reply:
P.S. the reason I ask if it's worth caching, is because `null` is a valid value - so to cache it I will need to add multiple additional fields.

Reply:
Maybe don't bother then, it is probably not a big deal.

Reply:
@Sanne Just for the record, you're not the only one avoiding lambdas in runtime code. We have some in build time but we really try to be careful for the runtime part.

But certainly some have slipped along the way.

Reply:
I can confirm what @gsmet mentions from my part as well.
We have instructed contributors multiple times to avoid them.
Of course some can sneak in 😁

Reply:
I mean there's plenty in the libraries we use - even if we avoid them in `io.quarkus.*` code ...

But I agree with the principle of not opting for a lambda without a strong reason, I'll be more careful. Thanks!